balena/models/application.py

```diff
@@ -382,24 +382,24 @@ def create(self, name, device_type, organization, app_type=None):
                 raise exceptions.OrganizationNotFound(organization)
 
         device_types = self.config.get_device_types()
-        device_manifest = [device for device in device_types if device['name'] == device_type]
+        device_manifest = [device for device in device_types if (device['name'] == device_type) or (device_type in device.get('aliases', []))]
 
         if device_manifest:
-            if device_manifest[0]['state'] == 'DISCONTINUED':
-                raise exceptions.BalenaDiscontinuedDeviceType(device_type)
-
-            params = {
-                'filter': 'slug',
-                'eq': device_manifest[0]['slug']
-            }
+            raw_query = "$filter=slug%20eq%20'{slug}'&$select=id,name&$expand=is_default_for__application($select=is_archived&$filter=is_host%20eq%20true)".format(slug=device_manifest[0]['slug'])
```

Comment:
```suggestion
            raw_query = "$filter=(slug%20eq%20'{slug}')%20or%20(name%20eq%20'{slug}')&$select=id,name&$expand=is_default_for__application($select=is_archived&$filter=is_host%20eq%20true)".format(slug=device_manifest[0]['slug'])
```
and make that request outside of the `if`
and change the `if` to
```
if not device_type_detail:
    raise exceptions.InvalidDeviceType(device_type)
```

Reply:
Updated, the request should be inside the `if ` since there will be exception if `device_manifest` is empty.

Reply:
I should have said
```
raw_query = "$filter=(slug%20eq%20'{device_type}')%20or%20(name%20eq%20'{device_type}')&$select=id,name&$expand=is_default_for__application($select=is_archived&$filter=is_host%20eq%20true)".format(device_type=device_type)
```
But w/o removing the `/device-types/v1` check, this doesn't have any effect and the second path of the `or` will never be true.

Feel free to revert to the original version of the query if you end up keeping the slug resolution via the `/device-types/v1` .